noah/src/main/scala/edu/uci/ics/cloudberry/noah/crawler/CrawlerDriver.scala

```diff
@@ -113,18 +125,16 @@ object CrawlerDriver extends App{
     }
 
     if(conf.hasPath("builder.crawlrules.clickelements")) {
-      val AIds = conf.getStringList("builder.crawlrules.clickelements.a.id")
       val AClasses = conf.getStringList("builder.crawlrules.clickelements.a.class")
-      val AText = conf.getStringList("builder.crawlrules.clickelements.a.text")
       val AHref = conf.getStringList("builder.crawlrules.clickelements.a.href")
+      val LiIds = conf.getString("builder.crawlrules.clickelements.li.id")
+      val InputIds = conf.getStringList("builder.crawlrules.clickelements.input.id")
+      val InputNames = conf.getStringList("builder.crawlrules.clickelements.input.name")
+      val InputValues = conf.getStringList("builder.crawlrules.clickelements.input.value")
 
-      var it = AIds.iterator()
-
-      while(it.hasNext){
-        builder.crawlRules().click("a").withAttribute("id", it.next())
-      }
+      builder.crawlRules().click("li").withAttribute("id", LiIds)
 
-      it = AClasses.iterator()
+      var it = AClasses.iterator()
```

Comment:
it's better to use `val` , event it may create more variables, which has no overhead for cpu but will make code more safe and readable by given them each a specific name.